otelconvert: panic when not using processor across all signals

### What's wrong?

While working at the config converters I came across a bug with the logic around NextMetrics/NextLogs/NextTraces.

I think it's a straightforward thing in `nextInPipeline`  where we have to check whether there are actually any processors before trying to look them up, but writing it down as an issue anyway in case it's not just that.

### Steps to reproduce

Use a config like the following on `otelconvert/testdata/batch2.yaml`.

```
receivers:
  otlp:
    protocols:
      grpc:
      http:

processors:
  batch:

exporters:
  otlp:
    # Our defaults have drifted from upstream, so we explicitly set our
    # defaults below (balancer_name and queue_size).
    endpoint: database:4317
    balancer_name: pick_first
    sending_queue:
      queue_size: 5000

service:
  pipelines:
    metrics:
      receivers: [otlp]
      processors: [batch]
      exporters: [otlp]
    logs:
      receivers: [otlp]
      processors: [batch]
      exporters: [otlp]
    traces:
      receivers: [otlp]
      processors: []
      exporters: [otlp]
```

Try to run the tests

```
$ go test ./... -run="TestConvert/batch2.yaml"
--- FAIL: TestConvert (0.00s)
    --- FAIL: TestConvert/batch2.yaml (0.00s)
panic: nextInPipeline: received processor ID not in processor list [recovered]
        panic: nextInPipeline: received processor ID not in processor list

goroutine 47 [running]:
testing.tRunner.func1.2({0x2715520, 0x34346a0})
        /usr/local/go/src/testing/testing.go:1545 +0x238
testing.tRunner.func1()
        /usr/local/go/src/testing/testing.go:1548 +0x397
panic({0x2715520?, 0x34346a0?})
        /usr/local/go/src/runtime/panic.go:914 +0x21f
github.com/grafana/agent/internal/converter/internal/otelcolconvert.nextInPipeline(0xc000aaa000, {{{0xc000625c00, 0x5}, {0x0, 0x0}}, 0x2, 0x0})
        /home/tpaschalis/GitRepos/agent/internal/converter/internal/otelcolconvert/pipeline_group.go:203 +0x727
github.com/grafana/agent/internal/converter/internal/otelcolconvert.pipelineGroup.NextTraces(...)
        /home/tpaschalis/GitRepos/agent/internal/converter/internal/otelcolconvert/pipeline_group.go:185
github.com/grafana/agent/internal/converter/internal/otelcolconvert.(*state).nextInstances(0xc000980060, {{{0xc000625c00, 0x5}, {0x0, 0x0}}, 0x2, 0x0}, {0x2d68a0d, 0x6})
        /home/tpaschalis/GitRepos/agent/internal/converter/internal/otelcolconvert/converter.go:174 +0x30a
github.com/grafana/agent/internal/converter/internal/otelcolconvert.(*state).Next(0xc000980060, {{{0xc000625c00, 0x5}, {0x0, 0x0}}, 0x2, 0x0}, {0x2d68a0d?, 0xc000a4f0b0?})
        /home/tpaschalis/GitRepos/agent/internal/converter/internal/otelcolconvert/converter.go:134 +0x72
github.com/grafana/agent/internal/converter/internal/otelcolconvert.toBatchProcessor(0x25?, {{{0xc000625c00, 0x5}, {0x0, 0x0}}, 0x2, 0x0}, 0xc000a523f0)
        /home/tpaschalis/GitRepos/agent/internal/converter/internal/otelcolconvert/converter_batchprocessor.go:49 +0x145
github.com/grafana/agent/internal/converter/internal/otelcolconvert.batchProcessorConverter.ConvertAndAppend({}, 0xc000980060, {{{0xc000625c00, 0x5}, {0x0, 0x0}}, 0x2, 0x0}, {0x283c560?, 0xc000a523f0})
        /home/tpaschalis/GitRepos/agent/internal/converter/internal/otelcolconvert/converter_batchprocessor.go:33 +0x145
github.com/grafana/agent/internal/converter/internal/otelcolconvert.appendConfig(0xc000aa8020, 0xc000ab6000)
        /home/tpaschalis/GitRepos/agent/internal/converter/internal/otelcolconvert/otelcolconvert.go:209 +0x122b
github.com/grafana/agent/internal/converter/internal/otelcolconvert.Convert({0xc0002fe000?, 0xc0001438c0?, 0xf?}, {0x4e8a1c0?, 0x24?, 0x5ad?})
        /home/tpaschalis/GitRepos/agent/internal/converter/internal/otelcolconvert/otelcolconvert.go:67 +0x4ca
github.com/grafana/agent/internal/converter/internal/test_common.TestDirectory.func1.1(0xc000602b60)
        /home/tpaschalis/GitRepos/agent/internal/converter/internal/test_common/testing.go:60 +0x2f5
testing.tRunner(0xc000602b60, 0xc0007a4a00)
        /usr/local/go/src/testing/testing.go:1595 +0xff
created by testing.(*T).Run in goroutine 46
        /usr/local/go/src/testing/testing.go:1648 +0x3ad
FAIL    github.com/grafana/agent/internal/converter/internal/otelcolconvert     0.021
```

### System information

_No response_

### Software version

_No response_

### Configuration

_No response_

### Logs

_No response_

--- a/internal/converter/internal/otelcolconvert/pipeline_group.go
+++ b/internal/converter/internal/otelcolconvert/pipeline_group.go
@@ -180,10 +180,6 @@ func nextInPipeline(pipeline *pipelines.PipelineConfig, fromID component.Instanc
 		// Processors should send to the next processor if one exists or to every
 		// exporter otherwise.
 		processorIndex := slices.Index(pipeline.Processors, fromID.ID)
-		if len(pipeline.Processors) > 0 && processorIndex == -1 {
-			panic("nextInPipeline: received processor ID not in processor list")
-		}
-
 		if processorIndex+1 < len(pipeline.Processors) {
 			// Send to next processor.
 			return []component.InstanceID{{Kind: component.KindProcessor, ID: pipeline.Processors[processorIndex+1]}}

--- a/internal/converter/internal/otelcolconvert/pipeline_group.go
+++ b/internal/converter/internal/otelcolconvert/pipeline_group.go
@@ -180,7 +180,7 @@ func nextInPipeline(pipeline *pipelines.PipelineConfig, fromID component.Instanc
 		// Processors should send to the next processor if one exists or to every
 		// exporter otherwise.
 		processorIndex := slices.Index(pipeline.Processors, fromID.ID)
-		if processorIndex == -1 {
+		if len(pipeline.Processors) > 0 && processorIndex == -1 {
 			panic("nextInPipeline: received processor ID not in processor list")
 		}
 

--- a/internal/converter/internal/otelcolconvert/testdata/inconsistent_processor.river
+++ b/internal/converter/internal/otelcolconvert/testdata/inconsistent_processor.river
@@ -1,0 +1,25 @@
+otelcol.receiver.otlp "default" {
+	grpc { }
+
+	http { }
+
+	output {
+		metrics = [otelcol.exporter.otlp.default.input]
+		logs    = [otelcol.processor.batch.default.input]
+		traces  = [otelcol.exporter.otlp.default.input]
+	}
+}
+
+otelcol.processor.batch "default" {
+	output {
+		metrics = [otelcol.exporter.otlp.default.input]
+		logs    = [otelcol.exporter.otlp.default.input]
+		traces  = [otelcol.exporter.otlp.default.input]
+	}
+}
+
+otelcol.exporter.otlp "default" {
+	client {
+		endpoint = "database:4317"
+	}
+}

--- a/internal/converter/internal/otelcolconvert/testdata/inconsistent_processor.yaml
+++ b/internal/converter/internal/otelcolconvert/testdata/inconsistent_processor.yaml
@@ -1,0 +1,33 @@
+receivers:
+  otlp:
+    protocols:
+      grpc:
+      http:
+
+processors:
+  batch:
+
+exporters:
+  otlp:
+    # Our defaults have drifted from upstream, so we explicitly set our
+    # defaults below (balancer_name and queue_size).
+    endpoint: database:4317
+    balancer_name: pick_first
+    sending_queue:
+      queue_size: 5000
+
+service:
+  pipelines:
+    metrics:
+      receivers: [otlp]
+      processors: []
+      exporters: [otlp]
+    logs:
+      receivers: [otlp]
+      processors: [batch]
+      exporters: [otlp]
+    traces:
+      receivers: [otlp]
+      processors: []
+      exporters: [otlp]
+

